### PR TITLE
Update docker images to RHEL8/GCC11 and MSVC 14.37

### PR DESCRIPTION
**Pull request by mistafunk**

### CODE REVIEW

File: envs/Jenkinsfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ psl.runsHere('production')
 
 @Field final String DOCKER_REGISTRY  = 'zrh-dreg-sp-1.esri.com'
 @Field final String DOCKER_PROJECT   = "${DOCKER_REGISTRY}/pyprt/"
-@Field final String DOCKER_IMAGE_REV = 'v9' // target revision of the generated images
+@Field final String DOCKER_IMAGE_REV = 'v10' // target revision of the generated images
```
Comment (clechot) [inline envs/Jenkinsfile:36]: I'm not sure but don't we need to update https://github.com/Esri/pyprt/blob/a959d140fc18c26682d5ee1216ba51f7815faa3e/Jenkinsfile#L41C41-L41C44 as well?

File: envs/windows/base/Dockerfile [hunk 2]
```diff
@@ -6,12 +6,12 @@ ARG BASE_IMAGE="mcr.microsoft.com/windows/servercore:1809"
...
 
-# see https://learn.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2019
-ADD https://aka.ms/vs/16/release/vs_buildtools.exe C:\temp\vs_buildtools.exe
+# see https://learn.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2022
+ADD https://aka.ms/vs/17/release/vs_buildtools.exe C:\temp\vs_buildtools.exe
 RUN C:\temp\vs_buildtools.exe --quiet --wait --norestart --nocache `
     --installPath C:\BuildTools `
     --add Microsoft.VisualStudio.Workload.VCTools --includeRecommended `
-    --add Microsoft.VisualStudio.Component.VC.14.27.x86.x64 `
+    --add Microsoft.VisualStudio.Component.VC.14.37.17.7.x86.x64 ^ `
```
Comment (clechot) [inline envs/windows/base/Dockerfile:14]: I see an additional `^`. Expected?

**Review by clechot — APPROVED** [review]
LGTM. Please review the two comments/questions I left.

File: envs/windows/base/Dockerfile (hunk 2 above)
Comment (mistafunk) [reply envs/windows/base/Dockerfile:14]: Eagle eye! I've aligned it with what we have in our [other containers](https://devtopia.esri.com/Zurich-R-D-Center/containers/blob/main/windows/vs-buildtools/Dockerfile.base#L49).

**Pull request merged**